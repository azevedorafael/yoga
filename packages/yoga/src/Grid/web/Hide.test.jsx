import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../..';
import { Hide } from '.';

describe('<Hide />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <ThemeProvider>
          <Hide xxs xs sm md lg xl xxl xxxl>
            <div>foo</div>
          </Hide>
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should hide starting from md breakpoint', () => {
      const { container } = render(
        <ThemeProvider>
          <Hide md-start>
            <div>foo</div>
          </Hide>
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
