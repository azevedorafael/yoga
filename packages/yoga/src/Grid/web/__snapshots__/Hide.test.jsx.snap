// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hide /> Snapshots should match snapshot 1`] = `
@media (max-width:360px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:360px) and (max-width:479px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:480px) and (max-width:767px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:768px) and (max-width:1023px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:1024px) and (max-width:1199px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:1200px) and (max-width:1365px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:1366px) and (max-width:1599px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:1600px) {
  .c0 {
    display: none !important;
  }
}

<div>
  <div
    class="c0"
  >
    <div>
      foo
    </div>
  </div>
</div>
`;
